docs/reference/contributing/target/QuadSPI.md

```diff
@@ -1,25 +1,50 @@
 <h3 id="quadspi-port">QuadSPI</h3>
 
-[Include a brief description here.]
+The QSPI HAL defines API for targets that contain QSPI capable peripheral. Developers often use the QSPI interface for data storage.
+
+<span class="warnings">**Warning:** We are changing the QSPI HAL API in an upcoming release of Mbed OS. You can find details on how it may affect you in the [Implementing the QSPI API](#implementing-the-qspi-api) section.
 
 #### Assumptions
 
 ##### Defined behavior
 
-[Include any defined behavior in bullet format here.]
+- A target implementaion covers most of the QSPI frame format (some targets might not provide the flexibility for setting all frame parameters).
+- Command transfer - A target might provide additional API for sending device-specific commands. If it does not, you can implement it using read and write functions. (This is target or driver dependent.)
 
 ##### Undefined behavior
 
-[Include any undefined behavior in bullet format here.]
+- Calling any function other than `qspi_init` before the initialization of the QSPI.
 
-##### Potential bugs
+#### Dependency
 
-[Include any potential bugs in bullet format here.]
+QSPI peripheral
 
 #### Implementing QuadSPI
 
-[Include implementation information here.]
+The target needs to define the `qspi_s` structure - target specific QSPI object.
+
+Functions to implement:
+
+```
+qspi_status_t qspi_init(qspi_t *obj, PinName io0, PinName io1, PinName io2, PinName io3, PinName sclk, PinName ssel, uint32_t hz, uint8_t mode);
+qspi_status_t qspi_free(qspi_t *obj);
+qspi_status_t qspi_frequency(qspi_t *obj, int hz);
+qspi_status_t qspi_write(qspi_t *obj, const qspi_command_t *command, const void *data, size_t *length);
+qspi_status_t qspi_command_transfer(qspi_t *obj, const qspi_command_t *command, const void *tx_data, size_t tx_size, void *rx_data, size_t rx_size); 
+qspi_status_t qspi_read(qspi_t *obj, const qspi_command_t *command, void *data, size_t *length);
+
+```
+
+Use `qspi_write` and `qspi_read` for data transfers. For communicating with a device, use `qspi_command_transfer`.
+
+To enable the QSPI HAL, define `QSPI` in the targets.json file inside `device_has`:
+
+```
+"TARGET_NAME": {
+   "device_has": ["QSPI"]
+}
+```
 
```

Comment:
Query: Is this on master? If so, please link to its Doxy on master. Then link to its Doxy on the feature branch. If not, please link only to the Doxy on the feature branch.

Reply:
Not on master, only on feature branch. how to link (an example) ?

Reply:
Ah. If it's not on master, then the link toward the top of this section is the correct one. I'm assuming any testing pages will be on the same branch, too: https://os-doc-builder.test.mbed.com/docs/development/feature-hal-spec-qspi-doxy/annotated.html?

We generate the feature-branch Doxy from the .json. You can find the page by replacing the "mbed-os-api-doxy" of the usual Doxy URL with the slug from the .json. (You probably don't need to know how to do that. I'm just telling you in case you're curious.)